pkg/utils/pfcp_util.go

```diff
@@ -11,31 +11,25 @@ import (
 	"github.com/free5gc/smf/pkg/service"
 )
 
-var (
-	pfcpStart func(a *service.SmfApp)
-	pfcpStop  func()
-)
-
 func InitPFCPFunc() (func(a *service.SmfApp), func()) {
-	pfcpStart = func(a *service.SmfApp) {
+	smfContext := smf_context.GetSelf()
+
+	pfcpStart := func(a *service.SmfApp) {
 		// Initialize PFCP server
-		ctx, cancel := context.WithCancel(context.Background())
-		smf_context.GetSelf().Ctx = ctx
-		smf_context.GetSelf().PFCPCancelFunc = cancel
+		smfContext.PfcpContext, smfContext.PfcpCancelFunc = context.WithCancel(context.Background())
```

Comment:
```go
smfContext.PfcpContext, smfContext.PfcpCancelFunc = context.WithCancel(pCtx)
```

Reply:
In this way,
the PfcpContext will be canceled once the app context is canceled.

Reply:
Makes sense